informer/informer.go

```diff
@@ -339,6 +350,52 @@ func (i *Informer) isCachedFilled() bool {
 	return false
 }
 
+func (i *Informer) newWatcher(ctx context.Context) (watch.Interface, error) {
+	var err error
+
+	var watcher watch.Interface
+	{
+		o := func() error {
+			failed := make(chan error, 1)
+			found := make(chan struct{}, 1)
+
+			go func() {
+				watcher, err = i.watcher.Watch(i.listOptions)
+				if err != nil {
+					failed <- microerror.Mask(err)
+					return
+				}
+
+				found <- struct{}{}
+			}()
+
+			select {
+			case <-ctx.Done():
+				return backoff.Permanent(microerror.Mask(contextCanceledError))
+			case err := <-failed:
+				return microerror.Mask(err)
+			case <-found:
+				// fall through
+			case <-time.After(time.Second):
+				return microerror.Mask(initializationTimedOutError)
+			}
+
+			return nil
+		}
+		b := backoff.NewExponentialBackOff()
```

Comment:
This probably has to be configurable. @MarcelMue and I already talked about having a general lib for this because we have quite some extensions here. Would maybe just help with this and do it here then. Let me know what you think.